src/base/bittorrent/filterparserthread.cpp

```diff
@@ -37,75 +42,120 @@
-                    else
-                        return false;
-
-                    if (endptr != str)
-                        return false;
-                    if (octetIndex == 4)
-                        return true;
-
-                    octetStart = str + 1;
-                }
-            }
+            // from_chars will stop when it encounters non-numeric chars
+            // we don't care for any other conversion failure
+            std::from_chars(pos, (line.data() + line.size()), entry.accessLevel);
+        }
+        else
+        {
+            // Apparently only range is mandatory
+            entry.range = line;
+        }
 
-            if (str != octetStart)
-            {
-                const long int extractedNum = strtol(octetStart, &endptr, 10);
-                if ((extractedNum >= 0L) && (extractedNum <= 255L))
-                    m_buf[octetIndex] = static_cast<unsigned char>(strtol(octetStart, &endptr, 10));
-                else
-                    return false;
-
-                if ((endptr == str) && (octetIndex == 3))
-                    return true;
-            }
+        return entry;
+    }
+
+    IPFilterEntry parseP2PLine(std::string_view line)
+    {
+        IPFilterEntry entry;
+
+        // Each line should follow this format:
+        // Some organization:1.0.0.0-1.255.255.255
+        // The "Some organization" part might contain a ':' char itself so we find the last occurrence
+        // Apparently this format wasn't designed for IPv6, so no effort will be made
+        // to parse possible IPv6 literals wrapped in []. These characters can appear in
+        // the "some organization" part too. It isn't worth implementing such complex parsing
+        // for a format that doesn't support IPv6 and there aren't any lists with IPv6 in them.
+        if (const auto colon = line.rfind(':'); colon != line.npos)
+            entry.range = line.substr(colon + 1);
+        else
+            entry.range = line;
+
+        return entry;
+    }
+
+    std::pair<std::string_view, std::string_view> parseRange(std::string_view range)
+    {
+        std::string_view start;
+        std::string_view end;
 
-            return false;
+        if (const auto pos  = range.find('-'); pos != range.npos)
+        {
+            start = range.substr(0, pos);
+            const auto pos_non_space  = std::find_if_not((range.data() + pos + 1), (range.data() + range.size()),
+                                              [](unsigned char c) { return std::isspace(c); } );
+            end = range.substr(pos_non_space - range.data());
         }
 
-        lt::address_v4::bytes_type parsed() const
+        return std::make_pair(start, end);
+    }
+
+    std::optional<lt::address> parseIPv4Address(std::string_view IPstr)
+    {
+        lt::address_v4::bytes_type octets = {};
+        int octetIndex = 0;
+
+        for (auto lastPos = IPstr.data();
+             lastPos < (IPstr.data() + IPstr.size());
```

Comment:
I don't object this but there are other lines longer than this for-loop so you don't need to fold it.